src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs

```diff
@@ -129,9 +135,20 @@ public ReadOnlyBytes Slice(int index)
             else {
                 var first = First;
                 if (first.Length > index) {
-                    return new ReadOnlyBytes(first.Slice(index), _rest);
+                    var slice = first.Slice(index);
+                    if(slice.Length > 0 || _rest!=null) {
+                        return new ReadOnlyBytes(slice, _rest);
```

Comment:
Yes, it should be an OR. If slice.length is zero but _rest is not null, we don't want to return an empty ROB. Same if slice is longer than zero.